pywbem/_logging.py

```diff
@@ -0,0 +1,402 @@
+  the info level. Internal calls to API functions are not logged.
+
+* 'pywbem.http' for http level information, at the info level.
+
+In addition, there may be loggers for pywbem modules with the module name, for
+situations like errors or warnings.
+
+All these loggers have a null-handler (see :class:`~py:logging.NullHandler`)
+and have no log formatter (see :class:`~py:logging.Formatter`).
+
+As a result, the loggers are silent by default. To turn on logging,
+add a log handler (see :meth:`~py:logging.Logger.addHandler`, and
+:mod:`py:logging.handlers` for the handlers included with Python) and set the
+log level (see :meth:`~py:logging.Logger.setLevel`, and :ref:`py:levels` for
+the defined levels).
+
+If you want to change the default log message format, use
+:meth:`~py:logging.Handler.setFormatter`. Its ``form`` parameter is a format
+string with %-style placeholders for the log record attributes (see Python
+section :ref:`py:logrecord-attributes`).
+
+Examples:
+
+* To output log records for one of the defined named loggers, either the
+  PywbemLogger.create_loggers or the Pywbem.create_logger can be used.
+
+  create_loggers creates one or more loggers from a string input that
+  defines the component name and characteristics of each logger.  This allows
+  other tools like clis to create the pywbem known logs with minimal work
+  from command line or config file input.
+
+  create_logger creates one or more loggers from parameter inputs that
+  define the component name and characteristics of each logger.
+
+* Also, loggers can be created by calling the logging  methods directly as
+   in this example.To output the log records for all cim operations to
+   ``stdout`` in a particular format, do this:
+
+  ::
+
+      import logging
+
+      handler = logging.StreamHandler()
+      format_string = '%(asctime)s - %(name)s - %(levelname)s - %(message)s'
+      handler.setFormatter(logging.Formatter(format_string))
+      logger = getLogger('pywbem.api')
+      logger.addHandler(handler)
+      logger.setLevel(logging.INFO)
+
+* This example uses the :func:`~py:logging.basicConfig` convenience function
+  that sets the same format and level as in the previous example, but for the
+  root logger. Therefore, it will output all log records, not just from this
+  package:
+
+  ::
+
+      import logging
+
+      format_string = '%(asctime)s - %(name)s - %(levelname)s - %(message)s'
+      logging.basicConfig(format=format_string, level=logging.DEBUG)
+"""
+from __future__ import absolute_import
+
+import logging
+import six
+
+# from ._constants import API_LOGGER_NAME
+API_LOGGER_NAME = 'pywbem.api'
+
+#: Name of logger for operantion call methods and their responses.
+#:
+LOG_OPS_CALLS_NAME = 'pywbem.ops'
+
+#: Name of logger for http information
+#:
+LOG_HTTP_NAME = 'pywbem.http'
+
+#: pywbem logger names. The following are the logger names that pywbem
+# recognizes.
```

Comment:
`#` -> `#:` and remove the empty line after the comment, to make the doc generator process this correctly.